src/main/java/com/mucommander/ui/autocomplete/TextFieldCompletion.java

```diff
@@ -77,6 +77,11 @@ public TextFieldCompletion(AutocompleterTextComponent comp, Completer completer)
 
     	autocompletedtextComp.getDocument().addDocumentListener(documentListener);
 
+        // In previous versions of muCommander, the 'home' and 'end' keys were forwarded
+        // to the pop-up list (if opened), which I found surprising. This version handles
+        // the 'home' and 'end' keys always in the text field, like Firefox and others do.
+        boolean SHOULD_HOME_AND_END_KEYS_WORK_ON_POPUP_LIST = false;
```

Comment:
Right, it may be different in the applications you mentioned. I checked Firefox and it is indeed different. On the other hand, Eclipse (for instance) behaves just like muCommander does at the moment. So I don't think there is a clear standard that we violate. In my opinion, we should find what is most appropriate for muCommander.

Now, since we don't provide an easy way to jump to the bottom of the popup list (such as pressing the 'up' key when the first item is selected, like in Firefox), I think that dropping the support for jumping straight to the first and last items in the list would be a step back from user experience. That is subjective, of course.

I don't see why would it be more difficult for you to make the change. You can change the code, commit it to the branch of the pull request on your forked repository and this pull request will be updated. This is a similar process to what I would do.